src/components/ContactList.tsx

```diff
@@ -1,7 +1,18 @@
 import { Link } from "react-router-dom";
+import { useState, useEffect } from "react";
 
 const ContactList = () => {
+    const [contacts, setContacts] = useState(null);
```

Comment:
It's more typical to set the default state to the same data type if possible, so empty array would be typical here

```suggestion
    const [contacts, setContacts] = useState([]);
```